### PR TITLE
Use monocart for coverage reports, enable codecov

### PR DESCRIPTION
**Pull request by jakebailey**
This is an experimental addition to `c8`, but works really well.

The v8 HTML page it outputs can load checker.ts without batting an eye, and has sub-expression fidelity and can see uncovered elses:

image

image

### CODE REVIEW

File: .c8rc.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "reporter": ["lcovonly", "cobertura"],
+    "reporter": ["lcovonly", "cobertura", "v8"],
```
Comment (jakebailey) [inline .c8rc.json:2]: I don't know how slow it is to always generate this; if it's expensive maybe we should just have a flag to customize the reporters we ask for through c8.

**Comment by jakebailey** [comment]
I'm going to revert the CI and codecov change until we can get faster runners to not double our CI time. The local thing is fine, though, so I'll leave that working. I'll also do a benchmark quick to see how the report collection changes.

**Comment by jakebailey** [comment]
The new runner pool is fast enough to give us coverage without slowing down CI.

File: .c8rc.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "reporter": ["lcovonly", "cobertura"],
+    "reporter": ["lcovonly", "cobertura", "v8", "codecov"],
```
Comment (jakebailey) [inline .c8rc.json:2]: Checked, and adding another reporter here does not affect the performance of the `report` stage, so we can keep this.

**Comment by jakebailey** [comment]
Marked this PR as ready for review. Now that the CI perf is mitigated, should we just go ahead and enable codecov?

**Comment by jakebailey** [comment]
A caveat is that c8 with monocart takes ~4x longer to run than c8 alone; reported https://github.com/cenfun/monocart-coverage-reports/issues/47 and the profile is somewhat promising for improving this.

**Review by RyanCavanaugh — APPROVED** [review]

**Pull request merged**